### PR TITLE
Fix for case 2 in #274

### PR DESCRIPTION
**Pull request by Herdo**
parser.GetCustomData() throws FileNotFoundException for missing DACPACs.

The FileNotFoundException is OK if the reference of the root DACPAC is not found.
It is however not OK if the missing DACPAC is only referenced by another DACPAC than the root DACPAC, but not by the root DACPAC directly.
Therefore, skip missing DACPACs if they're not directly referenced by the root DACPAC.

### CODE REVIEW

File: src/GUI/ReverseEngineer20/DacpacConsolidate/DacpacConsolidator.cs [hunk 1]
```diff
@@ -26,13 +26,21 @@ public string Consolidate(string dacpacPath)
...
+                // For that case, DACPACs not referenced directly by the root DACPAC are optional.
+                return Enumerable.Empty<string>().ToList();
+            }
+
             var parser = new HeaderParser(dacpacPath);
 
             var references = parser.GetCustomData()
-                .Where(p => p.Category == "Reference" && p.Type == "SqlSchema")
-                .ToList();
+                                   .Where(p => p.Category == "Reference" && p.Type == "SqlSchema")
```
Comment (ErikEJ) [inline src/GUI/ReverseEngineer20/DacpacConsolidate/DacpacConsolidator.cs:42]: Please fix indentation

**Comment by ErikEJ** [comment]
Looks good, assume you have verified? Just small formatting issue

File: src/GUI/ReverseEngineer20/DacpacConsolidate/DacpacConsolidator.cs (hunk 1 above)
Comment (Herdo) [reply src/GUI/ReverseEngineer20/DacpacConsolidate/DacpacConsolidator.cs:42]: Thank you ReSharper. Will fix.

**Comment by Herdo** [comment]
> Looks good, assume you have verified? Just small formatting issue

Currently on it - fails to build on my work machine, though.
I'll give you a GO when it works.

**Comment by ErikEJ** [comment]
Otherwise just fix the formatting, and I will run some testing after merge

**Comment by Herdo** [comment]
Okay, go ahead. **I'll try verify the merge with the new daily build.**
Local version is missing some manifest stuff I'm unable to find anywhere.

**Comment by ErikEJ** [comment]
Could you provide details about the build issue, it is important for me that contributors can build without friction?

(I will also update the efreveng.exe.zip file after merge, so hold on a little while with the daily build)

**Comment by ErikEJ** [comment]
Daily build 2.4.30 is now available

**Comment by Herdo** [comment]
@ErikEJ Build error on work machine was: `The target “GenerateFileManifest” does not exist in the project.`

I tried to verify the changes with the new daily build. However, the next error pops up now.
I'll need to debug this to solve that. Could you please have a look after the error above?

**Comment by ErikEJ** [comment]
@Herdo  Could you share your stack of files with me privately?
What is "the next error" ?
Re the error:
Do yo have the VS extensibility workload installed?
Which VS version do you use?

**Comment by ErikEJ** [comment]
@Herdo Ping?

**Comment by Herdo** [comment]
I got it to compile - currently working on the fix(es).

Visual Studio Version is 16.5.4. Visual Studio extension development workload is installed.
Building as x86 in Debug mode, with EFCorePowerTools as startup project. 

Now it’s working – error is gone. Only difference was restarting Visual Studio a few times.
No changes were made otherwise. Let’s just dismiss this as random Visual Studio error. 

I'll open two new pull requests for the fixes later.

**Comment by Herdo** [comment]
@ErikEJ Pong

**Comment by Herdo** [comment]
@ErikEJ Turns out "the next error" was a very weird state in several DACPACs.
It get's quiet complicated, and I conclude that there's no error in the current state of the EFCorePowerTools, instead our sqlprojects at work were messed up.

I'll make a new pull request with some details.

**Pull request merged**